### PR TITLE
chapter 11

### PR DESCRIPTION
**Pull request by mb706**

### CODE REVIEW

**Comment by mb706** [comment]
can mostly be reviewed, still waiting for one minor point from bernd

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 1]
```diff
@@ -165,6 +168,7 @@ tsk_adult
 ### Task {#sec-openml-task}
 
 OpenML tasks are built on top of OpenML datasets and additionally specify the target variable, the train-test splits to use for resampling, and more.
+Note that this differs from `mlr3` `Task` objects, also contain information about the target, but not about the resampling procedure.
```
Comment (RaphaelS1) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:171]:
```suggestion
Note that this differs from `mlr3` `Task` objects, which do not contain information about the resampling procedure.
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 2]
```diff
@@ -510,7 +517,7 @@ getStatus(reg = reg)
 ```
 
 To query the ids of jobs in the respective categories, see `r ref("findJobs()")` and, e.g., `r ref("findNotSubmitted()")` or `r ref("findDone()")`.
-In our case, we can see all experiments finished and none expired (`Expired : 0`) or crashed (`Error : 0`).
+In our case, we can see all experiments finished and none expired (i.e. were removed from the queue without ever starting, `Expired : 0`) or crashed (`Error : 0`).
```
Comment (RaphaelS1) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:520]:
```suggestion
In our case, we can see all experiments finished and none expired (i.e., were removed from the queue without ever starting, `Expired : 0`) or crashed (`Error : 0`).
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 3]
```diff
@@ -572,7 +579,7 @@ bmr$aggregate()[1:5]
 
 {{< include ../../common/_optional.qmd >}}
 
-In general, we recommend using `mlr3batchmark` for scheduling `mlr3` jobs on an HPC however we will also briefly show you how to use `batchtools` without `mlr3batchmark` for finer control over your experiment.
+In general, we recommend using `mlr3batchmark` for scheduling simpler `mlr3` jobs on an HPC however we will also briefly show you how to use `batchtools` without `mlr3batchmark` for finer control over your experiment.
```
Comment (RaphaelS1) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:582]:
```suggestion
In general, we recommend using `mlr3batchmark` for scheduling simpler `mlr3` jobs on an HPC however, we will also briefly show you how to use `batchtools` without `mlr3batchmark` for finer control over your experiment.
```

**Comment by RaphaelS1** [comment]
@mb706 looks good to me so far -- ping me when you're ready for me to do the final review

**Comment by RaphaelS1** [comment]
@mb706 please merge https://github.com/mlr-org/mlr3book/pull/711 into this if reverting back to "try"

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 4]
```diff
@@ -30,10 +30,10 @@ In @sec-openml we will begin by discussing `r ref_pkg("mlr3oml")`, which provide
...
 
-Throughout this chapter, we will use the running example of benchmarking a random forest model against a logistic regression (this was chosen as this problem has been extensively studied, e.g. @couronne2018random).
+Throughout this chapter, we will use the running example of benchmarking a random forest model against a logistic regression as in @couronne2018random.
 We will also assume that you have read @sec-pipelines and @sec-technical.
 We make use of `ppl("robustify")` (@sec-prepro-robustify) for automating common preprocessing steps.
-We also set a featureless baseline as a fallback learner (@sec-fallback) and set `"try"` as our encapsulation method, which logs errors/warnings to an external file that can be read by `batchtools` (we will return to this in @sec-batchtools-monitoring).
+We also set a featureless baseline as a fallback learner (@sec-fallback) and set `"evaluate"` as our encapsulation method, which logs errors/warnings to an external file that can be read by `batchtools` (we will return to this in @sec-batchtools-monitoring).
```
Comment (sebffischer) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:36]:
```suggestion
We also set a featureless baseline as a fallback learner (@sec-fallback) and set `"evaluate"` as our encapsulation method, which logs errors/warnings to an external file that can be read by `batchtools` (we will return to this in @sec-batchtools-monitoring).
```
```suggestion
We also set a featureless baseline as a fallback learner (@sec-fallback) and set `"try"` as our encapsulation method, which logs errors/warnings to an external file that can be read by `batchtools` (we will return to this in @sec-batchtools-monitoring).
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 5]
```diff
@@ -44,14 +44,14 @@ lrn_lr = lrn("classif.log_reg")
 lrn_lr = as_learner(ppl("robustify", learner = lrn_lr) %>>% lrn_lr)
 lrn_lr$id = "logreg"
 lrn_lr$fallback = lrn_baseline
-lrn_lr$encapsulate = c(train = "try", predict = "try")
+lrn_lr$encapsulate = c(train = "evaluate", predict = "evaluate")
```
Comment (sebffischer) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:47]:
```suggestion
lrn_lr$encapsulate = c(train = "try", predict = "try")
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 6]
```diff
@@ -44,14 +44,14 @@ lrn_lr = lrn("classif.log_reg")
...
 
 # random forest pipeline
 lrn_rf = lrn("classif.ranger")
 lrn_rf = as_learner(ppl("robustify", learner = lrn_rf) %>>% lrn_rf)
 lrn_rf$id = "ranger"
 lrn_rf$fallback = lrn_baseline
-lrn_rf$encapsulate = c(train = "try", predict = "try")
+lrn_rf$encapsulate = c(train = "evaluate", predict = "evaluate")
```
Comment (sebffischer) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:54]:
```suggestion
lrn_rf$encapsulate = c(train = "try", predict = "try")
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 7]
```diff
@@ -109,10 +109,13 @@ if (file.exists(path_odatasets)) {
...
 ```
 
+
 ```{r}
-odatasets[, c("data_id", "name")]
+odatasets[NumberOfFeatures < 16,
+  c("data_id", "name", "NumberOfFeatures", "NumberOfInstances")]
 ```
 
+Note that `odatasets()` returns a `data.table` with many more meta-features than shown here; this table can itself be used to filter further.
```
Comment (sebffischer) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:118]:
```suggestion
Note that `list_oml_data()` returns a `data.table` with many more meta-features than shown here; this table can itself be used to filter further.
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 8]
```diff
@@ -276,7 +280,8 @@ tasks = as_tasks(otask_collection)
 resamplings = as_resamplings(otask_collection)
 ```
 
-Alternatively, if we wanted to filter the collection further, say to a binary classification experiment with six tasks, we could run `r ref("list_oml_tasks()")` with the task IDs from the CC-18 collection as argument `task_id` and request the number of classes to be `2`.
+Alternatively, if we wanted to filter the collection further, say to a binary classification experiment with six tasks, we could run `r ref("list_oml_tasks()")` with the task IDs from the CC-18 collection as argument `task_id`.
+We can either use the `list_oml_task()` argument to request the number of classes to be `2`, or we can make use of the fact that the result of `list_oml_tasks()` is a `data.table` and subset the resulting table.
```
Comment (sebffischer) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:284]:
```suggestion
We can either use the `list_oml_tasks()` argument to request the number of classes to be `2`, or we can make use of the fact that the result of `list_oml_tasks()` is a `data.table` and subset the resulting table.
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 9]
```diff
@@ -572,7 +579,7 @@ bmr$aggregate()[1:5]
 
 {{< include ../../common/_optional.qmd >}}
 
-In general, we recommend using `mlr3batchmark` for scheduling `mlr3` jobs on an HPC however we will also briefly show you how to use `batchtools` without `mlr3batchmark` for finer control over your experiment.
+In general, we recommend using `mlr3batchmark` for scheduling simpler `mlr3` jobs on an HPC however, we will also briefly show you how to use `batchtools` without `mlr3batchmark` for finer control over your experiment.
```
Comment (sebffischer) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:582]:
```suggestion
In general, we recommend using `mlr3batchmark` for scheduling simpler `mlr3` jobs on an HPC, however, we will also briefly show you how to use `batchtools` without `mlr3batchmark` for finer control over your experiment.
```

File: book/chapters/chapter11/large-scale_benchmarking.qmd [hunk 10]
```diff
@@ -656,7 +664,7 @@ addExperiments(algo.designs = alg_des, reg = reg)
 summarizeExperiments()
 ```
 
-Our jobs can now be submitted to the cluster; by not specifying job IDs all experiments are submitted as independent jobs, i.e. one computational job executes one resample experiment.
+Our jobs can now be submitted to the cluster; by not specifying specific job IDs *all* experiments are submitted.
```
Comment (sebffischer) [inline book/chapters/chapter11/large-scale_benchmarking.qmd:667]:
```suggestion
Our jobs can now be submitted to the cluster; by not specifying specific job IDs, *all* experiments are submitted.
```

**Comment by RaphaelS1** [comment]
Thanks @sebffischer . No English edits from my side so will let @mb706 add the suggestions.
Just one comment from me above.

File: book/chapters/chapter11/large-scale_benchmarking.qmd (hunk 4 above)
Comment (RaphaelS1) [reply book/chapters/chapter11/large-scale_benchmarking.qmd:36]:
```suggestion
We also set a featureless baseline as a fallback learner (@sec-fallback) and set `"try"` as our encapsulation method (@sec-encapsulation), which logs errors/warnings to an external file that can be read by `batchtools` (we will return to this in @sec-batchtools-monitoring).
```

**Pull request merged**